### DISCUSSION TITLE
Do not automatically create preset when searching in a collection

### DISCUSSION THREAD

**Discussion by devistar** [directus/directus#15112] [Feature Requests]
### Describe the Bug

When searching in a collection, it automatically adds a record to the presets and favorites.
I think normally a record should be inserted only if clicked to create a preset.
Is this a bug?

### To Reproduce

Search a collection.

![image](https://user-images.githubusercontent.com/11228594/185070914-87efabf9-a317-4791-a457-d71aa7defca0.png)

**Comment by azrikahar** [comment]
> I think normally a record should be inserted only if clicked to create a preset.

It actually does also create a preset whenever you add/remove a field or column, sort column, change layout options etc. That being said, searching or filtering also creates a preset so that it will persist said search/filter which can be useful for some users 🤔 

Let's move this into a discussion to gather more feedback first 👍

**Comment by devistar** [comment]
I think providing an option to disable this behavior would be nice. 
This would avoid having to save a preset for each search performed. This would allow to have only the presets voluntarily saved.

**Comment by rijkvanzanten** [comment]
It's kinda important to keep filter/search persisted, otherwise the UX of navigating to an item and back, or to a different collection and back gets pretty janky (as the user would lose their search/location on the list 🤔 )

**Reply by Dominic-Marcelino** [reply]
That's absolutely true! Nevertheless I also do have a lot of problems with maintaining the presets for multiple users. 

I'd love to completely disable custom presets in some projects. Then every user has the same view on the data and whenever we change the preset it's applied for every user.

In this case the custom ones while filter/searching could be stores just in the localstorage and be session-based / easy to reset 🤔

Right now we're doing this by creating a 'default' preset any telling users to use this one instead of their own "unnamed" (when clicking on the collection name)

**Reply by devistar** [reply]
I think records for navigation have no place in the presets table.
It should either be in another table or in local storage.
Rather, it is more session than persistent data.
It would also be nice to leave the possibility of disable that.
The problem is that each search creates a record in the presets table and this pollutes the real presets that are saved.
As a solution, I put a check constraint to prevent creating a preset that has the bookmark column with a null value.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
